backend/impl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java

```diff
@@ -21,117 +21,108 @@
+        it.worktreeGitDirectoryPath);
 
-  @Test
-  @SneakyThrows
-  public void yieldsSameStatusAsCli() {
-    String gitMacheteCliStatus = gitMacheteCliStatusOutput();
+    String gitMacheteCliStatus = gitMacheteCliStatusOutput(scriptName);
 
-    BranchLayout branchLayout = branchLayoutReader.read(new FileInputStream(mainGitDirectoryPath.resolve("machete").toFile()));
-    gitMacheteRepositorySnapshot = gitMacheteRepository.createSnapshotForLayout(branchLayout);
-    String ourStatus = ourGitMacheteRepositorySnapshotAsString();
+    BranchLayout branchLayout = branchLayoutReader
+        .read(new FileInputStream(it.mainGitDirectoryPath.resolve("machete").toFile()));
+    val gitMacheteRepositorySnapshot = gitMacheteRepository.createSnapshotForLayout(branchLayout);
+    String ourStatus = ourGitMacheteRepositorySnapshotAsString(gitMacheteRepositorySnapshot);
 
     System.out.println("CLI OUTPUT:");
     System.out.println(gitMacheteCliStatus);
     System.out.println();
     System.out.println("OUR OUTPUT:");
     System.out.println(ourStatus);
 
-    Assert.assertEquals(gitMacheteCliStatus.trim(), ourStatus.trim());
+    assertEquals(gitMacheteCliStatus.trim(), ourStatus.trim());
+
+    cleanUpDir(it.parentDirectoryPath);
   }
 
-  @Test
   @SneakyThrows
-  public void discoversSameLayoutAsCli() {
-    String gitMacheteCliDiscoverOutput = gitMacheteCliDiscoverOutput();
+  @ParameterizedTest
+  @MethodSource("getScriptNames")
+  public void discoversSameLayoutAsCli(String scriptName) {
+    val it = new GitRepositoryBackedIntegrationTestSuiteInitializer(scriptName);
+    val gitMacheteRepository = gitMacheteRepositoryCache.getInstance(it.rootDirectoryPath, it.mainGitDirectoryPath,
+        it.worktreeGitDirectoryPath);
 
-    gitMacheteRepositorySnapshot = gitMacheteRepository.discoverLayoutAndCreateSnapshot();
-    String ourDiscoverOutput = ourGitMacheteRepositorySnapshotAsString();
+    String gitMacheteCliDiscoverOutput = gitMacheteCliDiscoverOutput(scriptName);
+
+    val gitMacheteRepositorySnapshot = gitMacheteRepository.discoverLayoutAndCreateSnapshot();
+    String ourDiscoverOutput = ourGitMacheteRepositorySnapshotAsString(gitMacheteRepositorySnapshot);
 
     System.out.println("CLI OUTPUT:");
     System.out.println(gitMacheteCliDiscoverOutput);
     System.out.println();
     System.out.println("OUR OUTPUT:");
     System.out.println(ourDiscoverOutput);
 
-    Assert.assertEquals(gitMacheteCliDiscoverOutput, ourDiscoverOutput);
-  }
+    assertEquals(gitMacheteCliDiscoverOutput, ourDiscoverOutput);
 
-  @Rule(order = Integer.MIN_VALUE)
-  public final TestWatcher cleanUpAfterSuccessfulTest = new TestWatcher() {
-    @Override
-    protected void succeeded(Description description) {
-      cleanUpDir(parentDirectoryPath);
-    }
-
-    // After a failed test, keep the parent directory intact for further manual inspection.
-  };
+    cleanUpDir(it.parentDirectoryPath);
```

Comment:
Okay... so basically, instead of a rule, just `cleanUpDir` at the end of each test, makes sense 😅 if the test fails at an earlier stage, it simply won't get to this point

Reply:
exactly

Reply:
Added a comment about that in code